sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java

```diff
@@ -1872,6 +1872,10 @@ public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource, int
 				getDao(Event.class).executeRaw("ALTER TABLE events ADD COLUMN connectionNumber varchar(512);");
 				getDao(Event.class).executeRaw("ALTER TABLE events ADD COLUMN travelDate timestamp;");
 
+			case 264:
+				currentVersion = 264;
+				getDao(Event.class).executeRaw("ALTER TABLE events RENAME surveillanceOfficer_id to responsibleUser_id;");
```

Comment:
I guess you haven't tested this? Renaming columns is not possible in the SQLite version used in Android. Our current approach is to rename the field in the entity class, but add the `name` attribute to reference the name in the database.

I have to emphasise again that it's vital to test the Android database migration.

Reply:
I did test it. I checked that the app worked properly and directly in database to see if the migration was applied correctly. And it was.

Reply:
SQLite supports alter table's column rename since 3.20 and android uses a superior version since API 28 (As the internet is telling me).
So, as a thoughtful development mastermind once said "It works on my machine".

What android's API level are we supposed to aim for ?

Reply:
We have to support API 24 (Android 7.0). It's specified in the main readme, but that's really easy to miss of course :) however, with API level 24 it should therefore not work, so we need the workaround that I've suggested.